### PR TITLE
Add frontend tests - part 2 (components - UIBox)

### PR DESCRIPTION
**Pull request by Dammic**
In this PR:
- added images to be parsed by identity-obj-proxy (so that tests don't throw errors about unexpected syntax on png files),
- moved store into separate file so that render (of whole app) is not executed on every test importing store,
- added missing key to GameEndModalComponent,
- Divided UIBox into 2 separate fully-tested components to make testing easier:
- -  GameControls - game control buttons,
- - Sidemenu - sidemenu with roles,
- ownerAction no longer need the required parameters passed - they take it from store now (it was A TODO)

### CODE REVIEW

File: src/features/UIBox/GameControls/__tests__/GameControlsComponent.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        gamePhase: null,
+        isAffiliationHidden: false,
+        ...propsOverrides,
+    }
+    const component = renderMethod(<GameControlsComponent {...props} />)
+    return { props, component }
+}
+
+describe('<GameControlsComponent />', () => {
+    it('renders right arrow if affiliation menu is hidden, and no owner buttons is user is not owner (and no start game button either)', () => {
```
Comment (Terverok) [inline src/features/UIBox/GameControls/__tests__/GameControlsComponent.test.js]: .. buttons **if user is not the owner** ..

File: src/features/UIBox/UIBox.jsx [hunk 2]
```diff
@@ -1,78 +1,15 @@
...
-        gamePhase: room.gamePhase,
-    }
-}
-
-const mapDispatchToProps = dispatch => ({
-    userActions: bindActionCreators({ toggleAffiliationMenu }, dispatch),
-})
-export default connect(mapStateToProps, mapDispatchToProps)(UIBox)
+UIBox.displayName = 'UIBox'
+export default UIBox
```
Comment (Terverok) [inline src/features/UIBox/UIBox.jsx]: Why not make it named export?

Comment (Terverok) [inline src/features/UIBox/UIBox.jsx]: Why not make it named export?

File: src/gameLogic/ownerActions.js [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 import { socket } from '../utils/SocketHandler'
 
-import { store } from '../AppClient'
+import { store } from '../store'
 import { toggleModal } from '../ducks/modalDuck'
 import * as playersActions from '../ducks/playersDuck'
 import * as roomActions from '../ducks/roomDuck'
 
-export function startGame(userName) {
-    socket.emit(SocketEvents.START_GAME, { playerName: userName })
```
Comment (Terverok) [inline src/gameLogic/ownerActions.js]: Why did you remove it?

File: src/gameLogic/ownerActions.js [hunk 4]
```diff
@@ -27,20 +28,25 @@ export function openInvitePlayersScreen() {
...
         playersActions.setChoiceMode({
             context: ChoiceModeContexts.KickChoice,
             selectablePlayers: playersWithoutOwner,
         }),
     ], store.dispatch)
 }
 
-export function startBanPlayerMode(userName, playersWithoutOwner) {
+export function startBanPlayerMode() {
+    const { user, room } = store.getState()
```
Comment (Terverok) [inline src/gameLogic/ownerActions.js]: :+1:

File: src/store.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        notification,
+        players,
+        modal,
+        lobby,
+        routing: routerReducer,
+    }),
+    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
+    compose(
+        applyMiddleware(thunk),
+        applyMiddleware(middleware),
```
Comment (Terverok) [inline src/store.js]: Instead of `middleware` we should name it `router`?

File: src/features/UIBox/UIBox.jsx [hunk 6]
```diff
@@ -1,78 +1,15 @@
 import React from 'react'
-import { connect } from 'react-redux'
-import { bindActionCreators } from 'redux'
-import { get, map, reject } from 'lodash'
-import UIBoxComponent from './UIBoxComponent'
-import PlayerAvatarComponent from '../PlayerBoard/Player/PlayerAvatar/PlayerAvatarComponent'
-import { toggleAffiliationMenu } from '../../ducks/userDuck'
-import { startGame, startKickPlayerMode, startBanPlayerMode, startVoting } from '../../gameLogic/ownerActions'
+import GameControls from './GameControls/GameControls'
+import Submenu from './Submenu/Submenu'
```
Comment (Terverok) [inline src/features/UIBox/UIBox.jsx]: Why can't we import these as named imports? :cry:

File: src/features/UIBox/UIBox.jsx (hunk 2 above)
Comment (Dammic) [reply src/features/UIBox/UIBox.jsx]: Named export you could have in index.js. Here it's the main part of the file you want to export, if we had more then sure.
For ex. you expect to import UIBox from UIBox.js (it's a component), but not setGamePhase from UIBox, that would be very confusing.
So if exporting something with the same name as the file, you would go for default export.

Comment (Dammic) [reply src/features/UIBox/UIBox.jsx]: Twice the same? ^

File: src/store.js (hunk 5 above)
Comment (Dammic) [reply src/store.js]: Let's go with routerMiddleware, shall we?

File: src/features/UIBox/UIBox.jsx (hunk 2 above)
Comment (Dammic) [reply src/features/UIBox/UIBox.jsx]: I'll add index.js for thise component to make importing better here.

File: src/store.js (hunk 5 above)
Comment (Dammic) [reply src/store.js]: nvm nvm let's go with router

File: src/features/UIBox/GameControls/__tests__/GameControlsComponent.test.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        gamePhase: null,
+        isAffiliationHidden: false,
+        ...propsOverrides,
+    }
+    const component = renderMethod(<GameControlsComponent {...props} />)
+    return { props, component }
+}
+
+describe('<GameControlsComponent />', () => {
+    it('renders right arrow if affiliation menu is hidden, and no owner buttons is user is not the owner (and no start game button either)', () => {
```
Comment (Terverok) [inline src/features/UIBox/GameControls/__tests__/GameControlsComponent.test.js]: ekhm.. !!  ... buttons **if** user is not ...
Damian, **if**.. **if** Damian.. :disappointed:
